ObjectTK.Tools/Cameras/Camera.cs

```diff
@@ -7,116 +7,70 @@
+		public Matrix4 ViewProjectionMatrix => ViewMatrix * ProjectionMatrix;
+		public Box2i Viewport { get; set; }
 
-        public float MouseMoveSpeed = 0.005f;
-        public float MouseWheelSpeed = 0.1f;
-        public float MoveSpeed = 60;
+		public float NearClippingPlaneDistance => 0.1f;
+		public float FarClippingPlaneDistance => 1000.0f;
 
-        public Camera()
-        {
-            State = new CameraState();
-            DefaultState = new CameraState();
-        }
+		public Matrix4 GetCameraTransform() {
+			// kind of hack: prevent look-at and up directions to be parallel
+			//if (Math.Abs(Vector3.Dot(Up, Forward)) > 0.99999999999) Forward += 0.001f * new Vector3(3, 5, 4);
+			return Matrix4.LookAt(Position, Position + Forward, Up);
+		}
 
-        public void ResetToDefault()
-        {
-            State.Position = DefaultState.Position;
-            State.LookAt = DefaultState.LookAt;
-            State.Up = DefaultState.Up;
-            Update();
-        }
+		public void Rotate(Vector3 EulerAngles) {
+			Quaternion Delta = (Matrix4.CreateFromAxisAngle(Right, EulerAngles.X) * Matrix4.CreateFromAxisAngle(Vector3.UnitY, EulerAngles.Y)).ExtractRotation();
+			Rotation = Delta * Rotation;
+		}
 
-        public void SetBehavior(CameraBehavior behavior)
-        {
-            Behavior = behavior;
-            Update();
-        }
+		public void LookAt(Vector3 Point) {
+			Rotation = Matrix4.LookAt(Position, Point, Up).ExtractRotation();
+		}
 
-        public void Enable(GameWindow window)
-        {
-            if (Behavior == null) throw new InvalidOperationException("Can not enable Camera while the Behavior is not set.");
-            window.UpdateFrame += UpdateFrame;
-            window.Mouse.Move += MouseMove;
-            window.Mouse.WheelChanged += MouseWheelChanged;
-        }
+		public Ray GetPickingRay(Vector2 MousePosition) {
 
-        public void Disable(GameWindow window)
-        {
-            window.UpdateFrame -= UpdateFrame;
-            window.Mouse.Move -= MouseMove;
-            window.Mouse.WheelChanged -= MouseWheelChanged;
-        }
+			Matrix4 UnViewProjectionMatrix = Matrix4.Invert(ViewProjectionMatrix);
 
-        public void Update()
-        {
-            if (Behavior != null) Behavior.Initialize(State);
-        }
+			Vector3 Near = Vector3.Unproject(
+				new Vector3(MousePosition.X, Viewport.Size.Y - MousePosition.Y, NearClippingPlaneDistance),
+				Viewport.Min.X,
+				Viewport.Min.Y,
+				Viewport.Size.X,
+				Viewport.Size.Y,
+				NearClippingPlaneDistance,
+				FarClippingPlaneDistance,
+				UnViewProjectionMatrix);
 
-        private void UpdateFrame(object sender, FrameEventArgs e)
-        {
-            Behavior.UpdateFrame(State, (float) e.Time * MoveSpeed);
-        }
+			Vector3 Far = Vector3.Unproject(
+				new Vector3(MousePosition.X, Viewport.Size.Y - MousePosition.Y, FarClippingPlaneDistance - NearClippingPlaneDistance),
```

Comment:
`FarClippingPlaneDistance - NearClippingPlaneDistance` is necessary for pre9.4 but hopefully my OpenTK PR gets merged and I can switch this to just `FarClippingPlaneDistance`

Reply:
Setting this to just FarClippingPlaneDistance with the expectation that this won't get merged at least until after pre9.5 which I hope will include my PR to fix it.